### PR TITLE
Migrate kubelet RootDirectory from flag to config

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Part of on-going Kubelet flags migration to config tasks.

**Which issue(s) this PR fixes**:

Ref: #86843

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
The Kubelet's `--root-dir` option is now available via the Kubelet config file field `RootDirectory`.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
none
```

/area kubelet
/area kubelet-api

### CODE REVIEW

File: pkg/kubelet/apis/config/helpers_test.go [hunk 1]
```diff
@@ -226,5 +226,6 @@ var (
 		"TypeMeta.Kind",
 		"VolumeStatsAggPeriod.Duration",
 		"VolumePluginDir",
+		"RootDirectory",
```
Comment (liggitt) [inline pkg/kubelet/apis/config/helpers_test.go:229]: this should be in the kubeletConfigurationPathFieldPaths list, *not* the kubeletConfigurationNonPathFieldPaths list

File: pkg/kubelet/apis/config/types.go [hunk 2]
```diff
@@ -136,6 +136,9 @@ type KubeletConfiguration struct {
 	// pulls to burst to this number, while still not exceeding registryPullQPS.
 	// Only used if registryPullQPS > 0.
 	RegistryBurst int32
+	// rootDirectory is the directory path to place kubelet files (volume
+	// mounts,etc).
+	RootDirectory string
```
Comment (liggitt) [inline pkg/kubelet/apis/config/types.go:141]: we already have VolumePluginDir and the existing flag is root-dir... suggest `RootDir`

File: staging/src/k8s.io/kubelet/config/v1beta1/types.go [hunk 3]
```diff
@@ -767,6 +769,11 @@ type KubeletConfiguration struct {
 	// Default: "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/"
 	// +optional
 	VolumePluginDir string `json:"volumePluginDir,omitempty"`
+	// rootDirectory is the directory path to place kubelet files (volume
+	// mounts,etc).
+	// Default: "/var/lib/kubelet"
+	// +optional
+	RootDirectory string `json:"rootDirectory,omitempty"`
```
Comment (liggitt) [inline staging/src/k8s.io/kubelet/config/v1beta1/types.go:776]: `RootDir`

File: pkg/kubelet/apis/config/v1beta1/defaults.go [hunk 4]
```diff
@@ -84,6 +84,9 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.Authorization.Webhook.CacheUnauthorizedTTL == zeroDuration {
 		obj.Authorization.Webhook.CacheUnauthorizedTTL = metav1.Duration{Duration: 30 * time.Second}
 	}
+	if obj.RootDir == "" {
+		obj.RootDir = kubeletconfigv1beta1.DefaultRootDir
+	}
```
Comment (liggitt) [inline pkg/kubelet/apis/config/v1beta1/defaults.go:89]: duplicate of lines 236-238?

File: pkg/kubelet/apis/config/v1beta1/defaults.go [hunk 5]
```diff
@@ -230,4 +233,7 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.VolumePluginDir == "" {
 		obj.VolumePluginDir = DefaultVolumePluginDir
 	}
+	if obj.RootDir == "" {
+		obj.RootDir = kubeletconfigv1beta1.DefaultRootDir
```
Comment (liggitt) [inline pkg/kubelet/apis/config/v1beta1/defaults.go:237]: should the `DefaultRootDir` const be in this package, peer to `DefaultVolumePluginDir`?

File: cmd/kubeadm/app/componentconfigs/kubelet.go [hunk 6]
```diff
@@ -173,6 +174,12 @@ func (kc *kubeletConfig) Default(cfg *kubeadmapi.ClusterConfiguration, _ *kubead
 		warnDefaultComponentConfigValue(kind, "healthzBindAddress", kubeletHealthzBindAddress, kc.config.HealthzBindAddress)
 	}
 
+	if kc.config.RootDir == "" {
+		kc.config.RootDir = kubeletconfigv1beta1.DefaultRootDir
+	} else if kc.config.RootDir != kubeletconfigv1beta1.DefaultRootDir {
+		warnDefaultComponentConfigValue(kind, "rootDir", kubeletconfigv1beta1.DefaultRootDir, kc.config.RootDir)
+	}
```
Comment (neolit123) [inline cmd/kubeadm/app/componentconfigs/kubelet.go:181]: you can omit this change from kubeadm given there is no need to override the default or print a warning if the user diverted from the default.

also, kubeletconfigv1beta1 imports the internal kubelet type which is not desired.

File: cmd/kubeadm/app/componentconfigs/kubelet_test.go [hunk 7]
```diff
@@ -188,6 +189,7 @@ func TestKubeletDefault(t *testing.T) {
 					Authorization: kubeletconfig.KubeletAuthorization{
 						Mode: kubeletconfig.KubeletAuthorizationModeWebhook,
 					},
+					RootDir:            kubeletconfigv1beta1.DefaultRootDir,
```
Comment (neolit123) [inline cmd/kubeadm/app/componentconfigs/kubelet_test.go:192]: see above. no need to include this in the unit tests too.

**Pull request closed without merging**